ewon_flexy_integration/blueprints/datasynchronization.py

```diff
@@ -0,0 +1,185 @@
+"""Scheduler for historic data synchronization"""
+import os
+import logging
+import base64
+from ewon_flexy_integration.utils.rest import TenantApi
+import requests
+from http.client import CONFLICT, NOT_FOUND, HTTPException
+from ewon_flexy_integration.blueprints.datamailbox import DataMailboxHandler
+from ewon_flexy_integration.models.c8y_ewon_flexy_integration import C8YEwonFlexyIntegration
+from c8y_api.model import ManagedObject
+
+from flask import Blueprint, jsonify, request
+from c8y_api.app import CumulocityApp, CumulocityApi
+
+bp = Blueprint('datasynchronization', __name__)
+logger = logging.getLogger('data synchronization')
+
+@bp.route('/executalljobs')
+def execute_all_jobs():
+    """Manually execute synchronization of all jobs via this endpoint.
+    """
+
+    dsh = DataSynchronizationHandler()
+    total_jobs_executed = dsh.synchronize_historic_data()
+    result_message = "Total jobs executed: " + str(total_jobs_executed)
+    logger.info(result_message)
+    return result_message
+
+
+@bp.route('/executejob')
+def execute_job():
+    """Manually execute synchronization for specific job via this endpoint.
+    """
+    token = request.headers.get('t2mtoken')
+    job_id = request.headers.get('jobId')
+    tenant_id = request.headers.get('tenantId')
+    dsh = DataSynchronizationHandler()
+    return dsh.syncdata(token, job_id, tenant_id)
+
+
+class DataSynchronizationHandler:
+
+    def __init__(self):
+        self.c8y = CumulocityApp(os.getenv('C8Y_BOOTSTRAP_TENANT'))
+
+    def synchronize_historic_data(self):
+        """Synchronize history data according to jobs. 
+        """
+        subscribed_tenants_list = self.get_subscribed_tenants_list()
+        if (len(subscribed_tenants_list) > 0):
+            for subscribed_tenant in subscribed_tenants_list:
+                subscribed_tenant_id = subscribed_tenant["tenant"]
+                subtenant_instance = self.c8y.get_tenant_instance(
+                    subscribed_tenant_id)
+                subtenant_response = subtenant_instance.get(
+                    "/inventory/managedObjects?pageSize=100&fragmentType=c8y_HMSOnloadingJob&currentPage=1&withTotalPages=true")
+                subtenant_jobs_mo_list = subtenant_response["managedObjects"]
+                jobs_executed = []
+                for job_mo in subtenant_jobs_mo_list:
+                    if (job_mo["isActive"]):
+                        owner = job_mo["owner"]
+                        category = self.create_tenant_option_category(owner)
```

Comment:
@yhegen this is where I fetch the token based on the owner's information in job mo.